Services/UI/OJS.Services.Ui.Business/Implementations/SubmissionsBusinessService.cs

```diff
@@ -152,10 +152,9 @@ public async Task<SubmissionDetailsServiceModel> GetDetailsById(int submissionId
             throw new BusinessServiceException(ValidationMessages.Submission.NotFound);
         }
 
-        var contest = await this.contestsDataService
-            .GetWithCategoryByProblem<ContestServiceModel>(submissionDetailsServiceModel!.Problem.Id);
-
-        var userIsAdminOrLecturerInContest = await this.lecturersInContestsBusiness.IsCurrentUserAdminOrLecturerInContest(contest!.Id);
+        var problem = await this.problemsDataService.GetWithProblemGroupById(submissionDetailsServiceModel.Problem.Id);
+        var userIsAdminOrLecturerInContest =
+            await this.lecturersInContestsBusiness.IsCurrentUserAdminOrLecturerInContest(problem!.ProblemGroup.ContestId);
```

Comment:
Don't you have ContestId in `submissionDetailsServiceModel`

Reply:
Fixed.